src/Microsoft.ML.TimeSeries/SsaAnomalyDetectionBase.cs

```diff
@@ -201,6 +210,38 @@ public sealed class State : AnomalyDetectionStateBase
             private SequenceModelerBase<Single, Single> _model;
             private SsaAnomalyDetectionBase _parentAnomalyDetector;
 
+            public State()
```

Comment:
>public State() [](start = 12, length = 14)

More public stuff...

So, do you see how I changed these things to be `private protected`? And then we introduce not one, not two, but *four* additional needlessly public?

I have to admit this is a little frustrating. I spend a considerable amount of effort trying to keep the codebase as clean as I can, but I can't do it myself.

Reply:
This needs to be public since State object is instantiated in SsaAnomalyDetectionBase.cs line 127

---
In reply to: [236828808](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236828808) [](ancestors = 236828808)

Reply:
Which is code in this assembly, so it can be internal. Not public.

---
In reply to: [236854833](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236854833) [](ancestors = 236854833,236828808)